keps/sig-network/4444-service-routing-preference/README.md

```diff
@@ -0,0 +1,998 @@
+  data plane will respect the Hints, it’s sufficient for the kube-proxy and
+  kube-controller-manager based implementations to say that we do support the
+  standard heuristics.
+
+### Changes within kube-proxy
+
+**Present behaviour:** kube-proxy only considers EndpointSlice hints for route
+programming if the `service.kubernetes.io/topology-aware-hints` annotation is
+set to "Auto"
+
+**New behaviour:** Irrespective of what the annotation
+`service.kubernetes.io/topology-aware-hints` or field `routingPreference` are
+set to (or even if they are not set at all), kube-proxy will always consider
+EndpointSlice hints (assuming this feature-gate is enabled).
+
+NOTE: The expectation remains that *all* endpoints within an EndpointSlice must
+  have corresponding hints for kube-proxy to utilize them. This avoids scenarios
+  with partial hints. The reason for this requirement is the same one highlighted in [KEP-2433 Topology
+  Aware
+  Hints](https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/2433-topology-aware-hints/README.md#kube-proxy), i.e. _"This is to provide safer transitions between enabled and disabled states. Without this fallback, endpoints could easily get overloaded as hints were being added or removed from some EndpointSlices but had not yet propagated to all of them."_
+
+### Status Reporting
+
+To provide clear status updates about the routing preferences to the user, the
+EndpointSlice controller (which is acting as the control plane) will update the
+Service status with the following conditions (inspired by Gateway API conditions)
+
+* RoutingPreferenceAccepted
+  * **Type:** `RoutingPreferenceAccepted` 
+  * **Description:** 
+    * Indicates whether a control plane component recognized and successfully
+      parsed the `routingPreference` value. A `False` status typically suggests
+      a configuration issue (e.g., an unsupported or malformed preference).
+    * The EndpointSlice controller will set this status to `True` if it
+      recognizes the `routingPreference` value as one it explicitly supports. 
+
+* RoutingPreferenceProgrammed
+  * **Type:**  `RoutingPreferenceProgrammed`
+  * **Description:** 
+    * This condition indicates whether they `routingPreference` has generated
+      some configuration that is assumed to be ready soon in the underlying data
+      plane.
+    * The EndpointSlice controller will set this status to `True` if it
+      successfully populated the EndpointSlice hints based on the
+      `routingPreference`.
+
+Note that the EndpointSlice controller and kube-proxy implementation does not
+currently provide a condition denoting acknowledgment from the dataplane. In the
+future when this is possible, another condition like the following could be
+used:
+
+* RoutingPreferenceReady
+  * **Type:** `RoutingPreferenceReady`
+  * **Description:** Confirms that the dataplane has received the hints,
+    acknowledged them, and configured itself accordingly.
```

Comment:
It'll be tricky to get right if we make that available - it requires aggregating the underlying status from lots of places.

---

I'd call this `RoutingPreferenceHonored`. Because it's a preference, not honoring it is OK - but people still want to know if it did get honored.